04.ls/ls.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+COL_COUNT = 3
+
+def main
+  entry_names = Dir.glob('*')
+  max_width = entry_names.map(&:size).max
+  shape_entry_names = shape_entry_names(entry_names)
+  puts_and_shape_width(shape_entry_names, max_width)
+end
+
+def shape_entry_names(entry_names)
+  slice_size = (entry_names.size / COL_COUNT.to_f).ceil
+
+  entry_names = entry_names.each_slice(slice_size).to_a
+  shape_entry_names = arrange_size(entry_names, entry_names[0].size)
+  shape_entry_names.transpose
```

Comment:
fillメソッドは配列の中身を破壊的に変更するので、メソッドを呼び出したあとに`entry_names`自身が変わってしまっています。
なので以下のようなコードでも動くはずです。

```suggestion
  arrange_size(entry_names, entry_names[0].size)
  entry_names.transpose
```

が、渡された引数を破壊的に変更するメソッドはあまりお行儀が良くないです。
https://bootcamp.fjord.jp/pages/do-not-use-destructive-method

若干消極的な対応ですが、`arrange_size`をメソッド化せずに、このmainメソッド内でfillすれば、まだ許容範囲かな〜という気がします。